tools/src/tensil/tools/compiler/Scheduler.scala

```diff
@@ -740,14 +773,25 @@ class Scheduler(
         accumulatorUtilization * 100f
       )
       tb.addNamedLine("Local utilization (%)", localUtilization * 100f)
-      tb.addNamedLine("True MACs (M)", macsCount.toFloat / 1e6f)
+      val (macsLetter, macsDivisor) =
+        BackendStats.getUnitsLetterAndDivisor(macs)
+      tb.addNamedLine(
+        s"True MACs (${macsLetter}MAC)",
+        macs.toFloat / macsDivisor
+      )
+      if (collectStats)
+        tb.addNamedLine("MAC efficiency (%)", macEfficiency * 100f)
       print(tb)
+    }
 
-      if (collectStats) {
-        BackendStats.printCompositionSummary(layerStats.get)
-        BackendStats.printCyclesSummary(layerStats.get)
-        BackendStats.printEnergySummary(layerStats.get)
+    if (collectStats) {
+      if (options.printInstructionsSummary) {
```

Comment:
It feels a bit odd to call this the "instructions summary" when it also gates the metrics. Probably fine for now but if you can think of a better name might be worth changing.

Reply:
Happy to change to something better. "Instructions summary" prints the following tables.

```
-----------------------------
TOTAL INSTRUCTIONS COMPOSITION (NUMBER/TOTAL SIZE)
-----------------------------
MatMul:                 2  2
DataMove(DRAM0ToLocal): 2  2
DataMove(DRAM1ToLocal): 2  34
DataMove(AccToLocal):   2  2
SIMD:                   2  2
DataMove(LocalToDRAM0): 2  2
LoadWeights:            2  34
TOTAL:                  14 78
-----------------------------
--------------------------------------
TOTAL INSTRUCTIONS LATENCY
--------------------------------------
MatMul:                 2.000   Cycles
DataMove(DRAM0ToLocal): 8.000   Cycles
DataMove(DRAM1ToLocal): 136.000 Cycles
DataMove(AccToLocal):   2.000   Cycles
SIMD:                   2.000   Cycles
DataMove(LocalToDRAM0): 4.000   Cycles
LoadWeights:            34.000  Cycles
TOTAL:                  188.000 Cycles
--------------------------------------
--------------------------------------
TOTAL INSTRUCTIONS ENERGY
--------------------------------------
MatMul:                 512.000 Units
DataMove(DRAM0ToLocal): 400.000 Units
DataMove(DRAM1ToLocal): 6.800   KUnits
DataMove(AccToLocal):   20.000  Units
SIMD:                   32.000  Units
DataMove(LocalToDRAM0): 400.000 Units
LoadWeights:            340.000 Units
TOTAL:                  8.504   KUnits
--------------------------------------
```

What will describe these better?

Reply:
Ah, my mistake, I somehow missed that this was the actual breakdown per instruction and not the overall metrics. Name is fine as it is!